core/arch/arm/kernel/secure_partition.c

```diff
@@ -53,9 +258,33 @@ static TEE_Result secure_partition_init(void)
 		else
 			msg[0] = '\0';
 		DMSG("SP %pUl size %u%s", (void *)&sp->uuid, sp->size, msg);
-	}
 
+		res = sp_init_uuid(&sp->uuid);
+
+		if (res != TEE_SUCCESS) {
+			EMSG("Failed initializing SP(%pUl) err: 0x%x"
+					, &sp->uuid, res);
+#ifndef SKIP_FAILED_SP
```

Comment:
Either rename to a `CFG_SP_xxx` or remove the condition.
If using a `CFG_xxx`, remember to use `if(IS_ENABLED(CFG_SP_xxx))` or like here.

I think whether optee panics when a user image panics maybe depends on the secure partition. Could be a config or flag in the SP image. But that's maybe not needed for now.

Reply:
At the moment there is nothing for this defined in the FF-A spec.  We also don't have any SP specific flags at the moment. I would keep it here for now and then change it later to a SP specific place.